### PR TITLE
Integration fixes for percona-server

### PR DESCRIPTION
**Pull request by yhr**

### CODE REVIEW

**Review by MaisenbacherD — COMMENTED** [review]
I have one comment, otherwise it looks good to me :)

File: fs/fs_zenfs.cc [hunk 1]
```diff
@@ -1383,9 +1383,10 @@ IOStatus ZenFS::MigrateFileExtents(
   IOStatus s = IOStatus::OK();
   Info(logger_, "MigrateFileExtents, fname: %s, extent count: %lu",
        fname.data(), migrate_exts.size());
+
   // The file may be deleted by other threads, better double check.
   auto zfile = GetFile(fname);
-  if (zfile == nullptr || zfile->IsOpenForWR()) {
+  if (zfile == nullptr || zfile->IsDeleted() || zfile->IsOpenForWR()) {
```
Comment (metaspace) [inline fs/fs_zenfs.cc:1388]: Maybe add a comment that the `IsDeleted` check is an optimistic optimization. Code is also valid without the check, but the check can help avoid redundant work.

**Review by metaspace — APPROVED** [review]
Looks good. Only have the one comment about the purpose of the `IsDeleted()` check when migrating extents.

File: fs/io_zenfs.cc [hunk 2]
```diff
@@ -446,6 +446,60 @@ IOStatus ZoneFile::AllocateNewZone() {
   return PersistMetadata();
 }
 
+/* Byte-aligned writes without a sparse header */
+IOStatus ZoneFile::BufferedAppend(char* buffer, uint32_t data_size) {
```
[the remark this answers is not in the document]
Comment (yhr) [reply fs/io_zenfs.cc:450]: Yes, I'm planning to do that as a follow up

File: fs/fs_zenfs.cc (hunk 1 above)
Comment (yhr) [reply fs/fs_zenfs.cc:1388]: Will do, i'll rework the commit message as well to clarify the isdeleted usage

Comment (metaspace) [reply fs/fs_zenfs.cc]: I think this was dropped, so resolving comment.

**Comment by aravind-wdc** [comment]
Looks good to me.

**Pull request merged**